libs/wingc/src/jsify.rs

```diff
@@ -41,21 +42,190 @@ const $App = __app(process.env.WING_TARGET);
+															property.name, 
+															self.method_name.name,
+														),
+														span: Some(property.span.clone()),
+													});
+													return;
+												}
+
+												// check if the type is capturable (resource, primitive or immutable collection)
+												if ! field_kind.type_.is_capturable() {
+													self.diagnostics.insert(Diagnostic {
+														level: DiagnosticLevel::Error,
+														message: format!(
+															"Unable to reference \"this.{}\" from inflight method \"{}\" because type {} is not capturable",
+															property.name, 
+															self.method_name.name,
+															field_kind.type_,
+														),
+														span: Some(property.span.clone()),
+													});
+
+													return;
+												}
+
+												debug!("field \"this.{}\" is referenced from inflight method {}", property.name, self.method_name.name);
+
+												// get/create an "ops" set for this field
+												let mut ops = if let Some(ops) = self.references.get(&property.name) {
+													ops.clone()
+												} else {
+													IndexSet::new()
+												};
+
+												if ! self.path.is_empty() {
+													let op = self.path[0].clone();
+
+													// if we are referencing a resource, verify that the operation is an inflight method
+													if let Some(r) = field_kind.type_.as_resource() {
+														let mut found_method = false;
+
+														for (name, typeref) in r.methods(true) {
+															if name == op {
+																match &*typeref {
+																	Type::Function(f) => {
+																		if f.flight == Phase::Inflight {
+																			found_method = true;
+																		}
+																	}
+																	_ => {}
+																}
+	
+															}
+														}
+
+														if !found_method {
+															self.diagnostics.insert(Diagnostic {
+																level: DiagnosticLevel::Error,
+																message: format!(
+																	"Unable to reference \"{}\" from inflight method \"{}\" because it is not an inflight method",
+																	self.path.join("."), 
+																	self.method_name.name,
+																),
+																span: Some(property.span.clone()),
+															});
+														}
+													}
+
+													ops.insert(op);
+												}
+			
+												self.references.insert(property.name.clone(), ops.clone());
+												self.path.clear();
+
+												// no need to visit recursively
+												return;
+											} else {
+												debug!("skipping field: {:?} because it is not in preflight", property.name)
+											}
+										}
+										_ => panic!("'this' is not a resource"),
+        					}
+
+								}
+							}
+							_ => {}
+						}
+					}
+					_ => {}
+    		}
+
+				self.path.insert(0, property.name.clone());
+			},
+			_ => {}
```

Comment:
Why are we not handling these cases? (_ => {}) (Maybe we can add comments saying which cases don't need to be handled by design and which are future TODOs, if possible)

Reply:
All of them don't need to be handled by design. Rewrote as simple `if let` statements to signal that this is all by design.